.github/workflows/build-and-test.yml

```diff
@@ -21,6 +22,68 @@ jobs:
         go-version: [1.22.x, 1.23.x]
         platform: [ubuntu-latest, macos-latest, windows-latest]
         feature-flags: ["DEFAULT", "PULUMI_TF_BRIDGE_ACCURATE_BRIDGE_PREVIEW"]
+        shard: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
+        exclude:
+          - platform: windows-latest
+            feature-flags: "PULUMI_TF_BRIDGE_ACCURATE_BRIDGE_PREVIEW"
+          - platform: macos-latest
+            feature-flags: "PULUMI_TF_BRIDGE_ACCURATE_BRIDGE_PREVIEW"
+          - platform: windows-latest
```

Comment:
Instead of defining `shard` as `[0, 1, 2, 3, ...]`, then excluding each option we don't want, can we define `shard` as `[0]` and then add explicit includes for other shard options. That will reduce the length of the shard list by half.